localstack/services/firehose/provider.py

```diff
@@ -779,3 +795,76 @@ def _get_s3_object_path(self, stream_name, prefix):
         path = pattern.format(pre=prefix, name=stream_name, rand=str(uuid.uuid4()))
         path = timestamp(format=path)
         return path
+
+    def _put_to_redshift(
+        self,
+        stream_name: str,
+        records: List[Dict],
+        redshift_destination_description: RedshiftDestinationDescription,
+    ):
+        jdbcurl = redshift_destination_description.get("ClusterJDBCURL")
+        cluster_id = self._get_cluster_id_from_jdbc_url(jdbcurl)
+        db_name = jdbcurl.split("/")[-1]
+        table_name = redshift_destination_description.get("CopyCommand").get("DataTableName")
+
+        columns_str = None
+        rows_to_insert = []
+        for record in records:
+            data = base64.b64decode(record.get("Data") or record.get("data"))
+            data = to_str(data)
+            data = json.loads(data)
+
+            if columns_str is None:
+                columns_str = ", ".join(data.keys())
+
+            values = []
+            for value in data.values():
+                if isinstance(value, str):
+                    # Escaping single quotes by doubling them
+                    value = value.replace("\t", " ")
+                    value = f"'{value}'"
+                elif value is None:
+                    value = "NULL"
+                else:
+                    value = str(value)
+                values.append(value)
+
+            value_str = ",".join(values)
+            rows_to_insert.append(value_str)
+
+        sql_insert_statement = f"INSERT INTO {table_name} ({columns_str}) VALUES "
+        for row in rows_to_insert:
+            sql_insert_statement += f"({row}),"
+        sql_insert_statement = sql_insert_statement.rstrip(",") + ";"
```

Comment:
We should pass the values in via the `Parameters` field rather than this SQL-injection-waiting-to-happen

Reply:
Since you have the two implementations, could you run some tests to work out how much slower the performance gets, and at what point it becomes prohibitive to use the `Parameters`?

Reply:
As mentioned this will only take effect if the firehose `_put_records` API is used to put multiple records to Firehose, this is not an AWS supported method to interact with Firehose directly. If Firehose receives messages from other services, each message is put individually to Firehose, therefore only a single row will be written each time. 

Considering this, `Parameters` are the better choice in the end.